test/NodeType.test.ts

```diff
@@ -45,41 +60,33 @@ describe('NodeType', () => {
   }
 
   beforeEach(() => {
-    context = {
-      httpApi: new TestHttpApi(),
-    };
-
-    const { nodeField, nodesField, ...rest } = createNodeType({
-      localIdFieldMode: 'deprecated',
-    });
-    ({ NodeType, createResolve } = rest);
+    // ({ NodeType, createResolve } = rest);
 
     class WidgetResource extends TestHttpResource {
       getFoo() {
         return this.api.foo();
       }
     }
 
-    const User = new NodeType({
+    const User: NodeType<WidgetResource> = new NodeType<WidgetResource>({
```

Comment:
this _should_ be inferable I think? if createResource returns T?

Reply:
see comment above

Reply:
not sure i understand isn't this: https://www.typescriptlang.org/play/?ssl=7&ssc=28&pln=1&pc=1#code/MYGwhgzhAEBiD28A8AVAfNA3gWAFDWmHgDsIAXAJwFdgz4KAKAMyuOAC5oGBKaAXgwpeOXHgIBfPJNG4ipMtACW-aMQCmAdziIGPfhgAs3ANxA

effectively the same thing but `i` here is properly inferred as `Foo<number>`

Reply:
I don't think your example applies. the constructor is not referencing itself. what am I missing?

Reply:
too make it more relevant: 

```ts
const Study = new NodeType({
  fields: {
    resolve: () => Study.getResource().get()
  }
});
```

Reply:
err... https://www.typescriptlang.org/play/#code/MYGwhgzhAEByD2ATApgFQJ4AdnQN4FgAoaaYeAOwgBcAnAV2CvhoAoAzAS2REQgC5oAJWRkaiADzUaHcgHMANHmg1kEeCABuyASwCU0ALwA+aGHLpoAXyP6ChIiUtEH0WcirC1dGsG2nzRE72hGSUVNAAylR0iBYG0OTIAO5wSGhYyCx2JJzcvALZJMqq6lo6+saR0bEAdG4eJd6+enXuei5BlroA3EA

Reply:
ugh, that's so dumb...

Reply:
seems like we should change the API _here_ then, if we are going to have re annotate every Node then there isn't much value to types here.

Reply:
I agree with that! but also, I think the factory is not that great...